### ISSUE TITLE
Send the ObjectMapper through the #apply method instead of giving it a JsonNode for output

### ISSUE THREAD

**Issue by luuse** [luuse/spock#10]
The `ObjectMapper` object is fairly expensive to create. How about adding a third argument to the `Trait#apply` method and friends that accepts the `ObjectMapper` we create in `Fetcher#main` but give to `Parser` which in turn calls `#apply`? This saves us from a few lines in each trait too...

This might become a bit messy in the interface for now but it's something we could easily clean up before delivery with a refactoring to have the `#apply` method return `null` instead of `false` and a non `null` value where we now return `true`.

**Comment by luuse** [comment]
This is partially fixed in 74a910cc074da0eaf3bbb411d54e2b4ec837f977. What's left is to remove the output node. However this raises the question, how should we merge nested objects?

**Comment by brodin** [comment]
Almost solved by https://github.com/luuse/spock/commit/e7bfaa310d85769dacfe2fb62078f558ee1fc98e

**Comment by brodin** [comment]
Two minor problems is left:
* How do we handle if multiple traits return? Or should we just have one parent trait? https://github.com/luuse/spock/blob/e7bfaa310d85769dacfe2fb62078f558ee1fc98e/parser/src/main/java/spock/Parser.java#L37
* Right now we can only return ´ObjectNode` and not arrays, look into how this can be improved https://github.com/luuse/spock/blob/e7bfaa310d85769dacfe2fb62078f558ee1fc98e/parser/src/main/java/spock/traits/CategoryTrait.java#L29

**Comment by brodin** [comment]
Must look over the documentation now.

**Comment by luuse** [comment]
I figured we could define it so that we run all root traits and merge their results? That way we could support other types of articles easily like organizations and it would make it easy to support articles describing a mix of things? This would solve #18 too.

Returning only ObjectNode is basically a feature of the above. It becomes a line or two extra but it's required to keep the traits separated and allow them to modify their entire subtree?

I've updated the documentation for the Trait class too so they describe the current state.
